File: relationship-manager/conversation-service/src/main/java/com/bank/rm/conversation/domain/Conversation.java
package com.bank.rm.conversation.domain;

import com.bank.rm.common.dto.*;
import jakarta.persistence.*;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "conversations")
public class Conversation {
    @Id
    private UUID id;

    @Column(name = "customer_id")
    private UUID customerId;

    @Column(name = "anonymous_session_id")
    private UUID anonymousSessionId;

    @Enumerated(EnumType.STRING)
    @Column(name = "current_phase")
    private ConversationPhase currentPhase;

    @Enumerated(EnumType.STRING)
    private ConversationStatus status;

    @Enumerated(EnumType.STRING)
    private CommunicationChannel channel;

    @Column(name = "extracted_data", columnDefinition = "jsonb")
    private String extractedData;

    @Column(name = "llm_context", columnDefinition = "jsonb")
    private String llmContext;

    @Column(name = "assigned_rm_id")
    private UUID assignedRmId;

    private String summary;

    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    protected Conversation() {}

    public Conversation(UUID id, UUID customerId, UUID anonymousSessionId, CommunicationChannel channel) {
        this.id = id;
        this.customerId = customerId;
        this.anonymousSessionId = anonymousSessionId;
        this.currentPhase = ConversationPhase.GREETING;
        this.status = ConversationStatus.ACTIVE;
        this.channel = channel;
        this.extractedData = "{}";
        this.llmContext = "{}";
        this.createdAt = Instant.now();
        this.updatedAt = Instant.now();
    }

    public UUID getId() { return id; }
    public UUID getCustomerId() { return customerId; }
    public UUID getAnonymousSessionId() { return anonymousSessionId; }
    public ConversationPhase getCurrentPhase() { return currentPhase; }
    public void setCurrentPhase(ConversationPhase currentPhase) { this.currentPhase = currentPhase; }
    public ConversationStatus getStatus() { return status; }
    public void setStatus(ConversationStatus status) { this.status = status; }
    public CommunicationChannel getChannel() { return channel; }
    public String getExtractedData() { return extractedData; }
    public void setExtractedData(String extractedData) { this.extractedData = extractedData; }
    public String getLlmContext() { return llmContext; }
    public void setLlmContext(String llmContext) { this.llmContext = llmContext; }
    public UUID getAssignedRmId() { return assignedRmId; }
    public String getSummary() { return summary; }
    public void setSummary(String summary) { this.summary = summary; }
    public Instant getCreatedAt() { return createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}

File: relationship-manager/conversation-service/src/main/kotlin/com/bank/rm/conversation/domain/Conversation.kt
package com.bank.rm.conversation.domain

import com.bank.rm.common.dto.*
import jakarta.persistence.*
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "conversations")
data class Conversation(
    @Id
    val id: UUID = UUID.randomUUID(),

    val customerId: UUID? = null,
    val anonymousSessionId: UUID? = null,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var currentPhase: ConversationPhase = ConversationPhase.GREETING,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var status: ConversationStatus = ConversationStatus.ACTIVE,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val channel: CommunicationChannel = CommunicationChannel.WEB,

    @Column(columnDefinition = "jsonb")
    var extractedData: String = "{}",

    @Column(columnDefinition = "jsonb")
    var llmContext: String = "{}",

    var assignedRmId: UUID? = null,
    var summary: String? = null,
    var lastMessageAt: Instant = Instant.now(),

    val createdAt: Instant = Instant.now(),
    var updatedAt: Instant = Instant.now()
)

@Entity
@Table(name = "messages")
data class Message(
    @Id
    val id: UUID = UUID.randomUUID(),

    @Column(nullable = false)
    val conversationId: UUID,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val senderType: SenderType,

    val senderId: UUID? = null,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val contentType: ContentType = ContentType.TEXT,

    @Column(columnDefinition = "TEXT", nullable = false)
    val content: String,

    @Column(columnDefinition = "jsonb")
    val metadata: String = "{}",

    @Enumerated(EnumType.STRING)
    val phase: ConversationPhase? = null,

    @Column(columnDefinition = "jsonb")
    val entitiesExtracted: String? = null,

    val createdAt: Instant = Instant.now()
)

@Entity
@Table(name = "conversation_templates")
data class ConversationTemplate(
    @Id
    val id: UUID = UUID.randomUUID(),

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, unique = true)
    val phase: ConversationPhase,

    val systemPrompt: String,

    @Column(columnDefinition = "jsonb")
    val requiredEntities: String = "[]",

    @Column(columnDefinition = "jsonb")
    val validationRules: String = "{}",

    val nextPhase: String? = null,
    val isActive: Boolean = true,
    val version: Int = 1
)
